plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -745,10 +752,9 @@ function ListMenu:_recalculateDimen()
     end
     local available_height = self.dimen.h - self.others_height
 
-    -- 64 hardcoded for now, gives 10 items both in filemanager
-    -- and history on kobo glo hd
-    local item_height_min = Screen:scaleBySize(64)
-    self.perpage = math.floor(available_height / item_height_min)
+    -- menu item height based on number of items per page
+    self.perpage = G_reader_settings:readSetting("files_per_page") or 10
+    local item_height_min = math.floor(available_height / self.perpage)
```

Comment:
Dunno if that, when keeping the default of 10, might change the computed item_height_min vs. the value before (if yes, the following issue will affect all users).
There's a small issue anyway when you change files_per_page: the cached thumbnails will be kept as rendered with the first value - so they may look a bit bad when you change that `files_per_page` value.
Not sure there's a way to invalide them and trigger a refresh - or if we let that to the user...

Or may be that's handled by `cover_specs` ? and if not, it could be included in sizetag/cover_sizetag, so you could use `"s10"` or `"s8"` or `"s14"` (when 10 or 8 or 14 files-per_page) instead of the simple `"s"`?

Reply:
When testing, the thumbnail sizes get adapted immediately, even on my old and slow Tolino Vision 2; so I don't think caching is an issue. Makes me wonder if the distinction between "w" and "s" is even necessary, or if we could just go with one size and have the device adapt it when displaying.

Reply:
Yes, they are resized to fit the wanted geometry if the saved blitbuffer has not the right size - but only in that case.

But if I took the pain of saving a rightly sized blitbuffer in the DB, it's to avoid that resize. Because if I introduced it, it's probably that I noticed some small slow down when it has to resize 9 or 10 small covers to a different small size, when you're browsing pages fast.
So, if you use only the `Detailed list` mode, we save the rightly sized small size blitbuffer (`"s"`)- and no resize needed when you quicly browse many file browser pages.
If you use Mosaic, we save bigger blitbuffers (`"M"`).

When you have both (Detailed for File browser, Mosaic for History, I think many people do that), we save the `"s"` when books are met in File Browser, but when meeting them in History, we notice if they have the `"s"` sizetag, and if they do, we refetch the original cover and resize it down to the bigger `"M" tag, and replace the blitbuffer in the DB, so they are nice in History (instead of a ugly "s" resized up to a bigger size). For quality.
Then, when back in File Browser, we may find some (those that we met once in History) with a "M" sizetag, and these indeed need to be resized down (on the fly, while browsing pages - we don't replace the M blitbuffer in the DB, we want to keep it for next time in History). There's a small slow down, but there shouldn't be too much books in History vs the amount we'll meet in File Browser.

So, not super necessary, but good to have and sad to drop :)

Reply:
I'll give it a go then, and use "s<num>" for indication. I'll try to keep using "s" when files_per_page is not set to prevent unnecessary resizing.

Reply:
My maths suck :) so I dunno if the bit of code I commented on here does change the computed `item_height_min`.
If there's a risk it does, you may just use `"s10"` for the default, so they are refreshed for all users next time they browse some area (so that the next times, they are the right size without any resize needed) - better that that a sudden feeling browsing has become a bit slower because of the 9-10 resizes for a few pixels more or less.
You could also use `"s<item_height_min>"` (and `"M<item_height_min>"`), so that next time we do some outer UI tweaks and happen to change `available_height `, that will be automatically taken care of.
Although `item_height_min` may also just change when you call history from portrait book, and then from a landscape book... so, well... dunno :/

Reply:
The height does change. On different size devices the 64 predefined height translates to a different number of files shown on one page.
Went with your suggestion, sizetag is now s<height_in_pixels>. I did not touch the "M" tag or size computation.